src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -619,6 +628,44 @@ private BoundExpression MakeImplicitConversion(BoundExpression rewrittenOperand,
                 rewrittenType: rewrittenType);
         }
 
+        private BoundExpression RewriteTupleConversion(
+            CSharpSyntaxNode syntax,
+            BoundExpression rewrittenOperand,
+            ConversionKind conversionKind,
+            bool @checked,
+            bool explicitCastInCode,
+            TupleTypeSymbol rewrittenType)
+        {
+            var destElementTypes = rewrittenType.TupleElementTypes;
+            var numElements = destElementTypes.Length;
+
+            var srcType = (TupleTypeSymbol)rewrittenOperand.Type;
+            var srcEleemntTypes = srcType.TupleElementTypes;
```

Comment:
Now I see your point, yes it looks like we should simply grab the type for the field. It would make the code clearer.

---
In reply to: [65627748](https://github.com/dotnet/roslyn/pull/11681#discussion_r65627748) [](ancestors = 65627748,65587124)